src/mpid/ch4/generic/am/mpidig_am_send.h

```diff
@@ -143,6 +149,17 @@ static inline int MPIDIG_do_ssend(const void *buf, MPI_Aint count, MPI_Datatype
     int mpi_errno = MPI_SUCCESS, c;
     MPIR_Request *sreq = *request;
 
+    MPL_pointer_type_t mem_type;
+    MPI_Aint data_sz;
+    MPL_gpu_query_pointer_type(buf, &mem_type);
+    MPIDI_Datatype_check_size(count, datatype, data_sz);
+    if (mem_type == MPL_GPU_POINTER_DEV && MPIDI_av_is_local(addr) &&
+        data_sz > mpidig_eager_limit(1)) {
+        /* GPU IPC protocol provides implicit synchronization for ssend */
+        return MPIDIG_gpu_ipc_send_req(buf, count, datatype, data_sz, rank, tag, comm,
+                                       context_offset, request, errflag);
+    }
```

Comment:
This is the same as `MPIDIG_gpu_lmt_send`, right? Can we test this case in `MPIDIG_isend_impl` and dispatch there? We need keep all dispatches together so we can easily read and refactor. Once we dispatched into a single protocol, that protocol need remain simple, do one thing and one thing only.